src/scraper/__init__.py

```diff
@@ -149,31 +298,76 @@ def scrape_main_page(self, base_url, odyssey_version, session, notes, logger, ms
             )
             main_soup = BeautifulSoup(main_page_html, "html.parser")
-            return main_page_html, main_soup
+        except Exception as e:
+            logger.error(f"Error scraping main page for main page HTML: {e}")
+            raise
+        return main_page_html, main_soup
         
-    def scrape_search_page(self, base_url, odyssey_version, main_page_html, main_soup, session, logger, ms_wait, court_calendar_link_text):
-        # build url for court calendar
+    def scrape_search_page(
+        self,
+        base_url: str,
+        odyssey_version: int,
+        main_page_html: str,
+        main_soup: BeautifulSoup,
+        session: requests.sessions.Session,
+        logger: logging.Logger,
+        ms_wait: int,
+        court_calendar_link_text: str
+    ) -> Tuple[str, str, BeautifulSoup]:
+        """
+        Scrapes the search page URL and data based on the main page content.
+
+        This method extracts the search page ID from the court calendar link, constructs the URL for the search page,
+        and retrieves the search page HTML. Depending on the Odyssey version, it either uses the extracted URL or a
+        default URL. It then parses the search page HTML into a BeautifulSoup object.
+
+        Args:
+            base_url (str): The base URL for constructing full URLs.
+            odyssey_version (int): The version of Odyssey, used to determine the correct URL and verification text.
+            main_page_html (str): The HTML content of the main page.
+            main_soup (BeautifulSoup): Parsed BeautifulSoup object of the main page HTML.
+            session (requests.sessions.Session): The session object for making HTTP requests.
+            logger (logging.Logger): Logger instance for logging errors and information.
+            ms_wait (int): Milliseconds to wait before making requests.
+            court_calendar_link_text (str): Text to search for in the court calendar link.
+
+        Returns:
+            Tuple[str, str, BeautifulSoup]: A tuple containing the search page URL, search page HTML, and the BeautifulSoup object of the search page.
+
+        Raises:
+            ValueError: If the court calendar link is not found on the main page.
+        """
+        # Extract the search page ID from the court calendar link
         search_page_id = None
         for link in main_soup.select("a.ssSearchHyperlink"):
             if court_calendar_link_text in link.text:
                 search_page_id = link["href"].split("?ID=")[1].split("'")[0]
+                break  # Exit loop once the link is found
+
         if not search_page_id:
             write_debug_and_quit(
                 verification_text="Court Calendar link",
                 page_text=main_page_html,
                 logger=logger,
             )
-        search_url = base_url + "Search.aspx?ID=" + search_page_id
+            raise ValueError("Court Calendar link not found on the main page.")
```

Comment:
Just highlighting that this is not actually using a default value if one isn't found, which isn't consistent with the docs

Reply:
Oof. I'll make sure to fix this. Good catch!

Reply:
Actually wait. Do you mean not using a default search_url or a default base_url?